### PR TITLE
feat(e2e-utils): add new api `useFixture`

### PR DESCRIPTION
**Pull request by Asuka109**
# PR Details

## Description

https://github.com/modern-js-dev/modern.js/blob/1da90a791630d7d92fa3e90c7c12bc8554a1d593/packages/toolkit/e2e/src/fixture.ts#L37-L44

https://github.com/modern-js-dev/modern.js/blob/ca3405d72af2b909c10f9e1fd68266255a9d7332/packages/toolkit/e2e/tests/fixture.test.ts#L6-L17

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/builder-webpack-provider/src/stub/builder.ts [hunk 1]
```diff
@@ -96,12 +96,13 @@ export async function createStubBuilder(options?: StubBuilderOptions) {
     options,
   ) as Required<StubBuilderOptions>;
   // apply webpack option.
-  if (options?.webpack) {
+  const config = builderOptions.builderConfig;
+  if (options?.webpack && !_.get(config, 'output.distPath.root')) {
```
Comment (jkzing) [inline packages/builder/builder-webpack-provider/src/stub/builder.ts:100]: 这里直接写 `config.output?.distPath?.root` 呗...

File: packages/builder/builder-webpack-provider/tests/e2e/fixture.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+it('fixture', async () => {
+  const options = await useFixture('@modern-js/e2e/fixtures/basic'); // also can be relative path
+  const builder = await createStubBuilder({
+    ...options,
+    plugins: 'minimal', // no need to set in e2e environment.
+  });
+  const files = await builder.unwrapOutputJSON();
+  expect(Object.keys(files)).toMatchInlineSnapshot(`
+    [
+      "<TEMP>/modern-js/stub-builder/dist/<FRAGMENT>/static/js/main.js",
```
Comment (jkzing) [inline packages/builder/builder-webpack-provider/tests/e2e/fixture.test.ts:14]:
```js
    "<TEMP>/modern-js/stub-builder/dist/<FRAGMENT>/static/js/main.js"
//  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
这一段的规则是啥，如果和 useFixture 的内部实现有关系的话可以整个用 `<TEMP>` 替换掉？

File: packages/toolkit/e2e/fixtures/basic/builder.fixture.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/** @type {import('@modern-js/builder-webpack-provider/stub').StubBuilderOptions} */
```
Comment (jkzing) [inline packages/toolkit/e2e/fixtures/basic/builder.fixture.js:1]: 这个路径里面是不是要加上 `builder`，`@modern-js/ede` 应该不只是给 builder 用的吧

比如 `@modern-js/e2e/fixtures/builder/basic` 或者 `@modern-js/e2e/fixtures/builder-basic`

File: packages/builder/builder-webpack-provider/tests/e2e/fixture.test.ts (hunk 2 above)
Comment (Asuka109) [reply packages/builder/builder-webpack-provider/tests/e2e/fixture.test.ts:14]: 这里的路径处理是快照序列化器提供的能力，temp 指的是系统临时目录的根级，配置在：
https://github.com/modern-js-dev/modern.js/blob/db25ef3457712b9416f05c7ea8018e82aa6bc7de/packages/builder/builder-webpack-provider/tests/setup.ts#L9-L20

File: packages/builder/builder-webpack-provider/src/stub/builder.ts (hunk 1 above)
Comment (Asuka109) [reply packages/builder/builder-webpack-provider/src/stub/builder.ts]: 🆗

File: packages/toolkit/e2e/src/fixture.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const pkgRoot = resolveFixturePackage(id);
+  let cwd = pkgRoot;
+  if (options?.copy) {
+    cwd = getTemplatePath('modern-js/stub-builder/e2e');
+    await fs.copy(pkgRoot, cwd, {
+      filter: isNonSymbolicLink,
+    });
+  }
+  const userOptions = resolveFixtureBuilderOptions(cwd);
+  const ret: Record<any, any> = {
```
Comment (jkzing) [inline packages/toolkit/e2e/src/fixture.ts:54]: 这个类型应该是 `StubBuilderOptions` ？

File: packages/builder/builder-webpack-provider/tests/e2e/fixture.test.ts (hunk 2 above)
Comment (jkzing) [reply packages/builder/builder-webpack-provider/tests/e2e/fixture.test.ts:14]: 那么 `<TEMP>/modern-js/stub-builder/dist/<FRAGMENT>` 这一段整个替换掉呗，测试中似乎不需要关注这段路径

File: packages/toolkit/e2e/src/fixture.ts (hunk 4 above)
Comment (Asuka109) [reply packages/toolkit/e2e/src/fixture.ts:54]: builder 引用了 e2e 包作为 devDep，这里再引用 stub builder options 的类型会触发循环依赖的检测，暂时先这么处理

File: packages/builder/builder-webpack-provider/tests/e2e/fixture.test.ts (hunk 2 above)
Comment (Asuka109) [reply packages/builder/builder-webpack-provider/tests/e2e/fixture.test.ts]: done.

File: packages/toolkit/e2e/fixtures/basic/builder.fixture.js (hunk 3 above)
Comment (Asuka109) [reply packages/toolkit/e2e/fixtures/basic/builder.fixture.js]: 确实，我改下

**Review by jkzing — APPROVED** [review]

**Pull request merged**